### DISCUSSION TITLE
Plugin examples that talk to a serial device?

### DISCUSSION THREAD

**Discussion by signal15** [Luligu/matterbridge#13] [Q&A]
I have a lighting controller with 48 dimmers and switches on it.  It has a serial port that I can send commands over, and poll each lighting load to find out it's status (it doesn't support unsolicited updates over serial, like when someone presses a button on the wall).  I wrote a plugin in TCL for the Vera which works fine with it, and I have a python program that implements a polling loop with a priority command queue.  

In any case, I want to get all of these lighting loads exposed via Matter so I can use them with Hubitat or Home Assistant.  Are there examples of plugins that talk to serial devices?

**Comment by Luligu** [comment]
Hi, that sounds very interesting.I have no news about it.If you write it let me know and we put it on the  Matterbridge page.

**Comment by Luligu** [comment]
If you need some specific apis let me know. I supported DIY devices in the matterbridge-zigbee2mqtt plugin.
